public/app/features/connections/tabs/ConnectData/CardGrid/CardGrid.tsx

```diff
@@ -0,0 +1,74 @@
+import { css } from '@emotion/css';
+import React, { FC } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Card, LoadingPlaceholder, useStyles2 } from '@grafana/ui';
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  sourcesList: css`
+    display: grid;
+    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
+    gap: 12px;
+    list-style: none;
+    margin-bottom: 80px;
+  `,
+  card: css`
+    height: 90px;
+    padding: 0px 24px;
+    box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.6);
+  `,
+  cardContent: css`
+    display: flex;
+    align-items: center;
+  `,
+  logoWrapper: css`
+    display: flex;
+    justify-content: center;
+    margin-right: 8px;
+    width: 32px;
+    height: 32px;
+    img {
+      max-width: 100%;
+      max-height: 100%;
+      align-self: center;
+    }
+  `,
+  label: css`
+    color: ${theme.colors.text.primary};
+    margin-bottom: 0;
+  `,
+});
+
+export interface CardGridProps {
+  items: Array<{ id: string; name: string; logo?: string }>;
+  isLoading: boolean;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  error: any;
+}
+
+export const CardGrid: FC<CardGridProps> = ({ items, isLoading, error }) => {
+  const styles = useStyles2(getStyles);
+
+  return isLoading ? (
+    <LoadingPlaceholder text="Loading..." />
+  ) : !!error ? (
+    <div>Error: {error.message}</div>
```

Comment:
As we are not using anything else (e.g. stack trace) from the error object apart from the `.message` field, it might makes sense to just have a `errorMessage?: string` prop instead of `error: any`. This could make the logic and the typing a bit simpler.

Thinking about it a bit more, maybe it's more the responsibility of `<ConnectData />` to handle any loading errors, as the async fetching actually happens there (and that is also the place where we are handling the case of no results). 

What do you think?

Reply:
The original reason why I put `error` in the `CardGrid` was:
> We'd like CardGrid to be reusable, even multiple times within a page.
In this case, it's better UX if we show the loading or error states per
card grid, not for the whole page.

But yeah, it actually feels like it doesn't belong there. And pages _could_ do something like: 
```
isCardGrid1Loading ? (
    <LoadingPlaceholder text="Loading..." />
  ) : !!errorCardGrid1 ? (
    <div>Error: {errorCardGrid1.message}</div>
) : <CardGrid one>

isCardGrid2Loading ? (
    <LoadingPlaceholder text="Loading..." />
  ) : !!errorCardGrid2 ? (
    <div>Error: {errorCardGrid2.message}</div>
) : <CardGrid two>
```
It feels like a lot of boilerplate, but actually it's better separation of concerns. :+1: